### PR TITLE
Add example App

### PR DESCRIPTION
**Pull request by Kicu**
## Description
This PR adds an example app to help developers who want to work locally on `react-native-qrcode-svg`.

All of the important code is kept within `Example` directory. This PR is expected to introduce no changes to the library or api, it just adds useful developer tooling

<p>
Screenshot 2024-08-23 at 12 10 50

</p>

https://github.com/user-attachments/assets/89d46e45-fdfb-4c2c-829d-a18afaa1ffdf

## Details
 - Example app built using expo
 - Example app has to be using `yarn` because when I used `npm` the local `link:...` dependency doesn't work; I have not found a better way to link to local library that wouldn't break also

### Supported platforms:
- ios
- android
- web
- macos

### Important files:
- Example/metro.config.js
- Example/webpack.config.js
- Example/package.json
- Example/src/App.tsx
- Example/README.md

### CODE REVIEW

**Comment by Kicu** [comment]
@luacmartins this is the PR that introduces Example app, it's mostly commits moved from our old fork, with some small tweaks so it works and builds correctly.
It's supposed to mimoc this PR: https://github.com/Expensify/react-native-qrcode-svg-old/pull/2


CC @grgia @awesomejerry (I have no idea if you're still contributing to this or not - feel free to comment if you do)

File: index.d.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@ export interface QRCodeProps {
   /* the color of the background */
   backgroundColor?: string;
   /* an image source object. example {uri: 'base64string'} or {require('pathToImage')} */
-  logo?: ImageSourcePropType;
+  logo?: ImageSourcePropType | string;
```
Comment (WoLewicki) [inline index.d.ts]: Why adding example app adds this change?

File: package.json [hunk 2]
```diff
@@ -47,11 +47,13 @@
     "react-native-svg": ">=13.2.0"
   },
   "dependencies": {
+    "@babel/runtime": "^7.25.4",
```
Comment (WoLewicki) [inline package.json]: Shouldn't it be rather `devDependency` ?

File: Example/package.json [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    "start": "react-native start",
+    "test": "jest",
+    "tsc": "tsc"
+  },
+  "dependencies": {
+    "@babel/plugin-proposal-export-namespace-from": "^7.18.9",
+    "@svgr/webpack": "^8.1.0",
+    "file-loader": "^6.2.0",
+    "react": "18.2.0",
+    "react-native": "^0.73.6",
```
Comment (WoLewicki) [inline Example/package.json:21]: Would be nice to bump the version to newest RN in some follow-ups. But then `macos` would probably stop working so idk.

**Review by WoLewicki — COMMENTED** [review]
Just some comments. Looks good overall 🚀

File: index.d.ts (hunk 1 above)
Comment (Kicu) [reply index.d.ts]: I will remove it 👍 you're right its confusing

**Review by luacmartins — APPROVED** [review]
LGTM

**Pull request merged**